web/app/charts/MapEditor.js

```diff
@@ -0,0 +1,457 @@
+        }
+        if(mapConfig !== null && mapConfig !== undefined){
+            Object.keys(mapConfig).forEach(val=>{
+                chartConfig[val] = mapConfig[val]? mapConfig[val]:null;
+            });
+            panel.title.innerHTML = mapConfig.chartTitle;
+            }
+        inputData = inputs;
+        createDropDown(optionsDiv);
+        createOptions();
+    };
+
+
+  //**************************************************************************
+  //** createDropDown
+  //**************************************************************************
+    var createDropDown = function(parent){
+
+        var table = createTable();
+        var tbody = table.firstChild;
+        table.style.height = "";
+        parent.appendChild(table);
+        createMapDropDown(tbody);
+    };
+
+  //**************************************************************************
+  //** createMapDropDown
+  //**************************************************************************
+    var createMapDropDown = function(tbody){
+        dropdownItem(tbody,"mapType","Map Type",showHideDropDowns,mapInputs,"mapType");
+        dropdownItem(tbody,"latitude","Latitude",createMapPreview,mapInputs,"lat");
+        dropdownItem(tbody,"longitude","Longitude",createMapPreview,mapInputs,"long");
+        dropdownItem(tbody,"mapValue","Value",createMapPreview,mapInputs,"mapValue");
+
+        var tr, td;
+
+        tr = document.createElement("tr");
+        tr.id="projection";
+        tbody.appendChild(tr);
+        td = document.createElement("td");
+        tr.appendChild(td);
+        td.innerHTML= "Projection:";
+
+        tr = document.createElement("tr");
+        tbody.appendChild(tr);
+        td = document.createElement("td");
+        tr.appendChild(td);
+
+        mapProjection = new javaxt.dhtml.ComboBox(td, {
+            style: config.style.combobox,
+            readOnly: true
+        });
+        mapProjection.clear();
+        mapProjection.onChange = function(name,value){
+            chartConfig.mapProjectionName = name;
+            chartConfig.mapProjectionValue = value;
+            createMapPreview();
+        };
+
+        document.getElementById("projection").style.visibility = "collapse";
+        mapProjection.hide();
+        document.getElementById("latitude").style.visibility = "collapse";
+        mapInputs.lat.hide();
+        document.getElementById("longitude").style.visibility = "collapse";
+        mapInputs.long.hide();
+        document.getElementById("mapValue").style.visibility = "collapse";
+        mapInputs.mapValue.hide();
+
+    };
+
+  //**************************************************************************
+  //** dropdownItem
+  //**************************************************************************
+    var dropdownItem = function(tbody,chartConfigRef,displayName,callBack,input,inputType){
+        var tr, td;
+
+        tr = document.createElement("tr");
+        tr.id = chartConfigRef;
```

Comment:
We should not be assigning an id. Nor should we be relying on document.getElementById(). It looks like the document.getElementById() is used to show/hide a row associated with an input. Here's a suggestion:
(1) addShowHide(tr);
(2) input[inputType].row = tr;

Then you can do something like this to hide the row:
mapInputs.mapValue.hide();
mapInputs.mapValue.row.hide();

Reply:
I have implemented this in my current branch.